app/actions/views/channel.js

```diff
@@ -259,8 +259,12 @@ export function selectInitialChannel(teamId) {
         const isGMVisible = lastChannel && lastChannel.type === General.GM_CHANNEL &&
             isGroupChannelVisible(myPreferences, lastChannel);
 
-        if (lastChannelId && myMembers[lastChannelId] &&
-            (lastChannel.team_id === teamId || isDMVisible || isGMVisible)) {
+        if (
+            lastChannelId &&
```

Comment:
`lastChannel` is being determined by `lastChannelId` so I think it is safe to remove it from the if statement if you are checking for `lastChannel`

Reply:
You're right, only that it's less explicit since inside the if-block, we're actually using the `lastChannelId`.  But it's all good, I think.
Updated.